sap/adt/dataelement.py

```diff
@@ -0,0 +1,185 @@
+"""ABAP Data Element ADT functionality module"""
+
+from sap.adt.objects import ADTObject, ADTObjectType, ADTCoreData, ADTObjectSourceEditor
+# pylint: disable=unused-import
+from sap.adt.annotations import OrderedClassMembers, xml_element, xml_text_node_property
+from sap.adt.objects import XMLNamespace, XMLNS_ADTCORE
+from sap.errors import SAPCliError
+
+XMLNS_DTEL = XMLNamespace('dtel', 'http://www.sap.com/adt/dictionary/dataelements')
+
+LABELS_LENGTH = {
+    'short': '10',
+    'medium': '20',
+    'long': '40',
+    'heading': '55'
+}
+
+
+class ADTDataElementData(ADTCoreData):
+    """Data Element nodes data.
+    """
+
+    # pylint: disable=too-many-instance-attributes
+    # pylint: disable=too-few-public-methods
+    class DataElement(metaclass=OrderedClassMembers):
+        """ADT Data Element data collector"""
+
+        type = xml_text_node_property('dtel:typeKind')
+        type_name = xml_text_node_property('dtel:typeName')
+        data_type = xml_text_node_property('dtel:dataType')
+        data_type_length = xml_text_node_property('dtel:dataTypeLength')
+        data_type_decimals = xml_text_node_property('dtel:dataTypeDecimals')
+        label_short = xml_text_node_property('dtel:shortFieldLabel')
+        label_short_length = xml_text_node_property('dtel:shortFieldLength')
+        label_short_max_length = xml_text_node_property('dtel:shortFieldMaxLength')
+        label_medium = xml_text_node_property('dtel:mediumFieldLabel')
+        label_medium_length = xml_text_node_property('dtel:mediumFieldLength')
+        label_medium_max_length = xml_text_node_property('dtel:mediumFieldMaxLength')
+        label_long = xml_text_node_property('dtel:longFieldLabel')
+        label_long_length = xml_text_node_property('dtel:longFieldLength')
+        label_long_max_length = xml_text_node_property('dtel:longFieldMaxLength')
+        label_heading = xml_text_node_property('dtel:headingFieldLabel')
+        label_heading_length = xml_text_node_property('dtel:headingFieldLength')
+        label_heading_max_length = xml_text_node_property('dtel:headingFieldMaxLength')
+        search_help = xml_text_node_property('dtel:searchHelp')
+        search_help_parameter = xml_text_node_property('dtel:searchHelpParameter')
+        set_get_parameter = xml_text_node_property('dtel:setGetParameter')
+        default_component_name = xml_text_node_property('dtel:defaultComponentName')
+        deactivate_input_history = xml_text_node_property('dtel:deactivateInputHistory')
+        change_document = xml_text_node_property('dtel:changeDocument')
+        left_to_right_direction = xml_text_node_property('dtel:leftToRightDirection')
+        deactivate_bidi_filtering = xml_text_node_property('dtel:deactivateBIDIFiltering')
```

Comment:
If you add these to the class DataElement, you can skip all those setter functions.